Modules/_abc.c

```diff
@@ -838,6 +693,62 @@ _abc_subclasscheck(PyObject *m, PyObject *args)
 }
 
 
+static int
+subclasscheck_check_registry(_abc_data *impl, PyObject *subclass,
+                             PyObject **result)
+{
+    Py_ssize_t registry_size = PySet_Size(impl->_abc_registry);
+    if (registry_size == 0) {
+        return 0;
+    }
+
+    int ret = 0;
+
+    // Weakref callback may remove entry from set.
+    // Se we take snapshot of registry first.
+    PyObject **copy = PyMem_Malloc(sizeof(PyObject*) * registry_size);
+    PyObject *key;
+    Py_ssize_t pos = 0;
+    Py_hash_t hash;
+    Py_ssize_t i = 0;
+
+    while (_PySet_NextEntry(impl->_abc_registry, &pos, &key, &hash)) {
+        Py_INCREF(key);
+        copy[i++] = key;
+    }
+    assert(i == registry_size);
+
+    for (i = 0; i < registry_size; i++) {
+        PyObject *rkey = PyWeakref_GetObject(copy[i]);
```

Comment:
copy[i] isn't necessary a weakref, if someone uses `weakref.getweakrefs` to access the internal registry, as I explained on the main PR. In this case, rkey would then be null.

Reply:
causing a crash in the Py_INCREF on the next line

Reply:
CPython is not sand-boxed language like JavaScript.
Basic operation and metaprogramming shouldn't cause crash.
But we don't have to guard from people break CPython via gc, ctypes, weakref, etc...

But in this case, adding `rkey == NULL` check is easy and natural.
I added it, thank you.